src/Microsoft.DotNet.Wpf/src/Themes/PresentationFramework.Win11/Styles/TabControl.xaml

```diff
@@ -0,0 +1,131 @@
+<!--
+    This Source Code Form is subject to the terms of the MIT License.
+    If a copy of the MIT was not distributed with this file, You can obtain one at https://opensource.org/licenses/MIT.
+    Copyright (C) Leszek Pomianowski and WPF UI Contributors.
+    All Rights Reserved.
+-->
+
+<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
+
+    <Style TargetType="{x:Type TabControl}">
+        <Setter Property="Foreground" Value="{DynamicResource TabViewForeground}" />
```

Comment:
I am concerned that a large number of DynamicResource declarations in this style might impact performance. Could we consider using StaticResource instead?

Reply:
The resources will change based on the theme, such as light, dark, or high contrast. Defining these resources as static will only bind them to one of these themes. We are still evaluating the performance impact of these changes. Do you have any suggestions to work around it?

Reply:
Resolving this for now as the issue is listed. Will keep things updating there,
Issue available [here.](https://github.com/dotnet/wpf/issues/8599)